File: src/main/java/org/example/lab3/SolutionString.java
package org.example.lab3;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class SolutionString {

    public static String countAndReplace(String s) {
        Pattern p = Pattern.compile(" o");
        Matcher m = p.matcher(s);
        int counter = 0;
        while (m.find()) {
            counter++;
        }
        System.out.println(counter);
        return s.replace(" o", "*");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Input first string: ");
        String s1 = scanner.nextLine();
        System.out.print("Input second string: ");
        String s2 = scanner.nextLine();
        System.out.println("Concatenation " + s1 + s2);
        System.out.println("Порівняння == : " + (s1==s2));
        System.out.println("Порівняння equals : "+s1.equals(s2));
        System.out.println("Порівняння compareTo : "+s1.compareTo(s2));



        s1 = countAndReplace(s1);

    }
}

File: lab1C/lab3/SolutionString.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace lab1C.lab3
{
    class SolutionString
    {
        //static void Main(string[] args)
        //{
        //    Console.Write("Input first string: ");
        //    String s1 = Console.ReadLine();
        //    Console.Write("Input first string: ");
        //    String s2 = Console.ReadLine();

        //    Console.WriteLine("Concatenation " + s1 + s2);

        //    Console.WriteLine("s1 == s2 : " + (s1 == s2));
        //    Console.WriteLine("s1.Equals(s2) : " + s1.Equals(s2));
        //    Console.WriteLine("s1.CompareTo(s2) : " + s1.CompareTo(s2));
        //    Console.WriteLine(countAndReplace(s1));
        //    Console.ReadLine();

        //}
        public static String countAndReplace(String s) {
            int counter = 0;
            int i = 0;
            while (i < s.Length - 1)
            {

                if (s[i] == ' ' && s[i + 1] == 'o')
                {
                    counter++;

                    i++;
                }
                i++;
            }
            s = s.Replace(" o", "*");
            Console.WriteLine(counter);
            return s;
        }
    }
}
